binnacle/lib/providers/navigation_provider.dart

```diff
@@ -13,30 +14,53 @@ class NavigationProvider {
   Vector2 _start;
   Vector2 _end;
   RouteModel _route;
-  BehaviorSubject eventBus = BehaviorSubject();
+  BehaviorSubject<NavigationEvent> eventBus = BehaviorSubject();
+  ReplaySubject<PositionModel> positionHistory;
+
+  List<LatLng> course;
 
   NavigationProvider({@required position, @required wind}) {
     _position = position;
     _wind = wind;
-    eventBus.add(NavigationEvent(eventType: NavigationEventType.awaitingInit));
-  }
-  initNavigation() {
-    print('!!! FROM INITNAV');
+
     eventBus.add(NavigationEvent(eventType: NavigationEventType.init));
   }
 
-  // start(PositionModel start, PositionModel end) {
-  //   _start = _positionToCartPoint(start);
-  //   _end = _positionToCartPoint(end);
+  start(LatLng start, LatLng end) {
+    /// NOTE: naive to assume long = x, lat = y, but it's close enough
+    /// A better implementation would utilize the Haversine formula
+    /// Which IMO should be done in the algorithm
+    _start = Vector2(start.longitude, start.latitude);
+    _end = Vector2(end.longitude, end.latitude);
 
-  // }
-  // RouteModel calculateRoute(start, end) {
+    eventBus
+        .add(NavigationEvent(eventType: NavigationEventType.calculatingRoute));
 
-  // }
+    _route = new RouteModel(start: _start, end: _end, wind_radians: 0);
+    // new RouteModel(start: _start, end: _end, wind_radians: currentWind.deg);
 
-  // _positionToCartPoint(p) => CartPoint(p.lat, p.lon);
-  // _cartPointToPosition(c) => PositionModel(lat: c.x, lon: c.y, speed: 0.0);
+    course = List();
+    course.add(start);
+    _route.intermediate_points.forEach((point) {
+      course.add(new LatLng(point.y, point.x));
+    });
+    course.add(end);
 
+    print('Route generated');
+    print(_route);
+
+    eventBus.add(NavigationEvent(eventType: NavigationEventType.start));
+
+    positionHistory = new ReplaySubject<PositionModel>();
+    positionHistory.addStream(_position);
+
+    // TODO: Nick when producing the finish event, we need to run this line
+    // positionHistory.close();
+
+    // TODO: Convert this to a stack implementation
+    // We don't want this clear to be pushed if a new event arrived in the bus
+    Future.delayed(Duration(seconds: 5), () => eventBus.add(NavigationEvent(eventType: NavigationEventType.init)));
```

Comment:
Is this a good practice? If so, we might want to notify the user when the navigation actually starts.

Reply:
This is just a hack to clear the navigation panel at the top. Otherwise it will always say starting course. Future work requires user story 2 to be done.

Reply:
Understood.